.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,55 @@
+name: pull request
+
+on: [pull_request, workflow_dispatch]
+
+jobs:
+  code_checks:
+    name: Code checks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+      - name: Set Node Version
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.14.2
+      - name: Install deps
+        run: npm ci
+      - name: Code Checks
+        run: npm run prepublishOnly
+
+  publish_npm_betas:
```

Comment:
With the current triggers for this workflow, this job will run on every pull request and each update to each pull request. Typically a beta release only gets published on commits to a designated branch, like `beta`, or `next` for example. Would this be preferable?
I don't think we should be publishing a new beta release on every PR, it may get too noisy.

Reply:
yeah it might be a lot. inspired by [connect](https://github.com/hirosystems/connect/runs/6451469565?check_suite_focus=true). it's great for testing out pr changes with _actual_ npm install, rather than local linking. the releases are tagged under a `pr` dist-tag and the version includes the git sha.

i'll remove since it's likely not needed in this repo